main.py

```diff
@@ -63,20 +65,52 @@ def enrich_report(report, columns):
     return report
 
 
-def build_report(source, schema):
+def validate(source, schema):
     report = validata_core.validate(source, schema)
     columns = report["tables"][0]["headers"]
 
     return enrich_report(report, columns)
 
 
+def build_report(report):
+    def badge_url(nb_errors, color):
+        query = urlencode(
+            {
+                "label": "Consolidation",
+                "message": f"{nb_errors} erreurs",
+                "color": color,
+                "style": "flat-square",
+            }
+        )
+        # See documentation on https://shields.io
+        return f"https://img.shields.io/static/v1?{query}"
+
+    percentage = int(report["nb_errors"] * 100 / report["nb_rows"])
+    if percentage == 0:
+        status, color = "ok", "green"
+    elif percentage <= 10:
+        status, color = "warning", "orange"
+    else:
+        status, color = "invalid", "red"
+
+    return {
+        **report,
+        **{
+            "status": status,
+            "error_percentage": percentage,
+            "badge_url": badge_url(report["nb_errors"], color),
+        },
+    }
+
+
 def build_details(details, report):
     errors = report["tables"][0]["error-stats"]
 
     return {
-        "date": datetime.date.today(),
+        "date": str(datetime.date.today()),
```

Comment:
Needed for JSON serialization. CSV serialization dumps `datetime`s in ISO 3601, not JSON. Too bad

Reply:
`.isoformat()`?